### PR TITLE
Add multiple route options to reducer

### PR DESCRIPTION
**Pull request by el-mapache**
This pull request add support for multiple routes in the route option reducer. Each `routeOptionView` gets passed the index of the route it is responsible for.

## Additions

- Add a route factory function to build new route data objects

## Removals

-

## Changes

- Since the tool will only allow for two routes for now, explicitly initialize
each view and expandable in `index.js`
- Update reducer to handle multiple route objects

## Testing

1. Just unit tests at this point

## Notes

- I don't love all the manual selecting of `expandables` and `views` in the `index.js` file, but we only need two route options for the tool and this is the simplest way to ensure that all the data goes to the right places and all the JS objects get initialized correctly

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [ ] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: cfgov/unprocessed/apps/youth-employment-success/js/reducers/route-option-reducer.js [hunk 1]
```diff
@@ -26,6 +28,46 @@ const updateDailyCostAction = actionCreator(
...
+/**
+ *
+ * @param {array} routes the array of routes in the reducer's state
+ * @param {number} routeIndex the index of the route object we want to update
+ * @param {object} data object describing the updates we want to make
+ *  to the route object's proeprties
+ * @returns {object} the next state of the reducer
+ */
+function updateRouteData( routes, routeIndex, data ) {
+  return {
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/youth-employment-success/js/reducers/route-option-reducer.js]: If you have a big block of logic you're returning I might store it as an object and then return that, which I think is a little clearer when parsing over the code, like:

```
const nextState = {
  routes: routes.map( ( route, index ) => {
    if ( index !== routeIndex ) {
      return route;
    }

    return assign( route, data );
  } )
};

return nextState;
```

File: cfgov/unprocessed/apps/youth-employment-success/js/route-option-view.js [hunk 2]
```diff
@@ -43,29 +44,35 @@ function RouteOptionFormView( element, { store } ) {
    * @param {object} updateObject object with DOM event and field name
    */
   function _setQuestionResponse( { name, event } ) {
-    const method = actionMap[name];
+    const action = actionMap[name];
+    const { target: { value }} = event;
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/youth-employment-success/js/route-option-view.js]: TRIVIAL: should be `const { target: { value } } = event;` looks like

File: cfgov/unprocessed/apps/youth-employment-success/js/route.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { assign } from './util';
+
+/**
+ * Factory function for creating a route object. Designed to be used in conjunction
+ * with the routeOptionReducer
+ * @param {object} route optional object describing the values of the route object
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/youth-employment-success/js/route.js]: You can enclose the parameter name in brackets to designate to jsdocs that it's optional (see [here](https://devhints.io/jsdoc)). So should be:

```
* @param {Object} [route] optional object describing the values of the route object.
```

File: cfgov/unprocessed/apps/youth-employment-success/js/route-option-view.js (hunk 2 above)
Comment (el-mapache) [reply cfgov/unprocessed/apps/youth-employment-success/js/route-option-view.js]: I wonder why the linter didn't catch this! 😠

File: cfgov/unprocessed/apps/youth-employment-success/js/route.js (hunk 3 above)
Comment (el-mapache) [reply cfgov/unprocessed/apps/youth-employment-success/js/route.js]: oh cool, thanks!

**Pull request merged**